Atomic/scan.py

```diff
@@ -178,50 +177,70 @@ def set_scanner():
             return (json.dumps(self.get_scan_data()))
 
     def _get_scan_list(self):
-
-        def gen_images():
-            slist = []
-            for image in self.get_images():
-                if image['ImageType'] == 'system':
-                    image['Id'] = image['RepoTags'][0]
-                image['input'] = image['Id']
-                slist.append(image)
-            return slist
-
-        def gen_containers():
-            slist = []
-            for con in self.get_containers():
-                # If con does not have 'Status' return empty string
-                if con.get('Status', '').upper() != 'DEAD':
-                    con['input'] = con['Id']
-                else:
-                    raise ValueError("The container ID {} is Dead.".format(con['Id'][:12]))
-                slist.append(con)
-            return slist
-
+        beu = BackendUtils()
         if self.args.images:
-            scan_list = gen_images()
+            scan_list = beu.get_images()
         elif self.args.containers:
-            scan_list = gen_containers()
+            scan_list = beu.get_containers()
         elif self.args.all:
-            scan_list = gen_containers() + gen_images()
+            scan_list = beu.get_images() + beu.get_containers()
         else:
             scan_list = []
-            images = self.get_images()
-            containers = self.get_containers()
-            for scan_input in self.args.scan_targets:
-                docker_object = (next((item for item in containers
-                                       if item['Id'] == self.get_input_id(scan_input)), None))
-                docker_object = docker_object if docker_object is not None \
-                    else (next((item for item in images if item['Id'] == self.get_input_id(scan_input)), None))
-                if docker_object is not None:
-                    docker_object['input'] = scan_input
-                    scan_list.append(docker_object)
-            if len(scan_list) < 1:
-                raise ValueError("You must provide at least one container or image for atomic "
-                                 "scan. See 'atomic scan --help' for more information")
-
+            for scan_target in self.args.scan_targets:
+                try:
+                    # get_backend_and_container throws a ValueError when it cannot find anything
+                    _, scan_obj = beu.get_backend_and_container_obj(scan_target)
```

Comment:
could we rewrite it as `_, scan_list = beu.get_backend_and_container_obj(scan_target)`

and move the second try/except block to the same level as the first one?  The only other change required is:
```
if scan_list != []:
   _, scan_list = beu.get_backend_and_image_obj(scan_target)
```

What do you think?